### PR TITLE
[SDK-1379] Export constructor

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Description

Add the option to use the `Auth0Client` constructor without having to use the factory.

The Existing API stays the same:

```javascript
// ES Modules
import createAuth0Client from '@auth0/auth0-spa-js';
// CommonJS
const createAuth0Client = require('@auth0/auth0-spa-js')
// Browser
window.createAuth0Client;
```

Now you can also access the [Auth0Client constructor](https://auth0.github.io/auth0-spa-js/classes/auth0client.html) directly:

```javascript
// ES Modules
import createAuth0Client, { Auth0Client } from '@auth0/auth0-spa-js';
// CommonJS
const { createAuth0Client, Auth0Client } = require('@auth0/auth0-spa-js')
// Browser
window.createAuth0Client;
window.Auth0Client;
```

If you access the `Auth0Client` constructor directly, it's up to you to check the user's session to automatically log the user in.

```javascript
import { Auth0Client } from '@auth0/auth0-spa-js';

const auth0 = new Auth0Client({ clientId, domain });
await auth0.getTokenSilently().catch(e => {
  if (e.error_message !== 'login_required') throw e;
});
```

### Testing

- [x] The integration tests check the browser behaviour
- [x] The Unit tests check the ES Module behaviour
- [x] Manually check the Common JS case
- [x] Test against [Vue sample](https://github.com/auth0-samples/auth0-vue-samples)
- [x] Test against [React sample](https://github.com/auth0-samples/auth0-react-samples/)
- [x] Test against [Angular sample](https://github.com/auth0-samples/auth0-angular-samples)
- [x] Test against [JavaScript sample](https://github.com/auth0-samples/auth0-javascript-samples)

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,6 @@
   "version": "1.7.0-beta.4",
   "main": "dist/lib/auth0-spa-js.cjs.js",
   "types": "dist/typings/index.d.ts",
-  "browser": "dist/auth0-spa-js.production.js",
```
Comment (damieng) [inline package.json]: Why is this removed?

File: rollup.config.js [hunk 2]
```diff
@@ -34,13 +34,16 @@ const getPlugins = shouldMinify => {
     sourcemaps()
   ];
 };
+const footer = `('Auth0Client' in this) && this.console && this.console.warn && this.console.warn('Auth0Client already declared on the global namespace');
```
Comment (damieng) [inline rollup.config.js]: I don't understand the changes to this file in context of the PR description - should this be something separate?

File: package.json (hunk 1 above)
Comment (adamjmcgrath) [reply package.json]: Webpack resolves the library by checking the package properties in this order: "browser" -> "module" -> "main"

This is designed so that if you have a library that provides a different flavour for nodejs and the browser (like `axios`)

We only have one flavour, browser, but we do want to provide an optimized es module version to webpack (or any build tool that prefers es modules). So if we delete "browser" webpack will use module and get the benefits of esm (tree shaking, named exports), things that use commonjs (like browserify) will use "main"

File: rollup.config.js (hunk 2 above)
Comment (adamjmcgrath) [reply rollup.config.js]: This exposes the constructor on the window object (as `window.Auth0Client`)

Similar to how Marionette provides the alias `window.Mn` https://github.com/marionettejs/backbone.marionette/blob/master/rollup.config.js#L28

Comment (stevehobbsdev) [reply rollup.config.js]: I believe the idea here is to provide a warning in the browser console when there is already an `Auth0Client` symbol defined on the window object, as including our library would then try to override it.

@adamjmcgrath can you remind me of the behaviour here? Does it override it, or do we *not* override it in favour of the object that was already there?

Comment (adamjmcgrath) [reply rollup.config.js]: It doesn't override `Auth0Client` if that property already exists on the `window`, it'll raise a warning if this is the case.
